pkg/alertmanager/multitenant.go

```diff
@@ -901,6 +901,7 @@ func (am *MultitenantAlertmanager) newAlertmanager(userID string, amConfig *defi
 		Retention:                         am.cfg.Retention,
 		MaxConcurrentGetRequestsPerTenant: am.cfg.MaxConcurrentGetRequestsPerTenant,
 		ExternalURL:                       am.cfg.ExternalURL.URL,
+		TmplExternalURL:                   tmplExternalURL,
```

Comment:
The external URL can change when a new config is applied. For example, consider the following scenario:
1. A user has no Grafana configuration, they are using Mimir's external URL
2. They upload some Grafana configuration, they should be using Grafana's external URL

We should assign the correct external URL on every config sync (check [here](https://github.com/grafana/mimir/blob/369932771fb4eacef7209b4fd5046965af999513/pkg/alertmanager/alertmanager.go#L518-L520), we're updating the templates in the Alertmanager struct, we could use the same lock to update the external URL).

Reply:
Ah yeah, ofc.. I missed the `else` and presumed a new instance would be built every time. Tks, fixed now 👍